### PR TITLE
Consistency with signature verification messages

### PR DESCRIPTION
**Pull request by castulo**
This commit enhance the consistency and readability, of the messages
shown when the signature verification is skipped or fails.

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

### CODE REVIEW

File: src/signature.c [hunk 1]
```diff
@@ -241,43 +242,45 @@ bool signature_verify(const char *file, const char *sig_file, bool print_errors)
 	/* get the signature */
 	sig_fd = open(sig_file, O_RDONLY);
 	if (sig_fd == -1) {
-		string_or_die(&errorstr, "Failed open %s: %s\n", sig_file, strerror(errno));
+		string_or_die(&errorstr, "Failed open %s: %s", sig_file, strerror(errno));
```
Comment (karthikprabhuvinod) [inline src/signature.c]: Failed to open

File: src/signature.c [hunk 2]
```diff
@@ -241,43 +242,45 @@ bool signature_verify(const char *file, const char *sig_file, bool print_errors)
...
-		string_or_die(&errorstr, "Failed to mmap %s signature\n", sig_file);
+		string_or_die(&errorstr, "Failed to mmap %s signature", sig_file);
 		goto error;
 	}
 	/* get the data to be verified */
 
 	data_fd = open(file, O_RDONLY);
 	if (data_fd == -1) {
-		string_or_die(&errorstr, "Failed open %s\n", file);
+		string_or_die(&errorstr, "Failed open %s", file);
```
Comment (karthikprabhuvinod) [inline src/signature.c]: Failed to open

File: src/signature.c [hunk 3]
```diff
@@ -444,12 +446,12 @@ static int validate_certificate(X509 *cert, const char *certificate_path, const
 
 	/* create a verification context and initialize it */
 	if (!(verify_ctx = X509_STORE_CTX_new())) {
-		error("Failed X509_STORE_CTX_new() for %s\n", certificate_path);
+		debug("Failed X509_STORE_CTX_new() for %s\n", certificate_path);
```
Comment (karthikprabhuvinod) [inline src/signature.c]: why are we changing these from error to debug

File: src/version.c [hunk 4]
```diff
@@ -176,10 +176,9 @@ static int get_version_from_url(char *url)
 	if (globals.sigcheck) {
 		sig_verified = verify_signature(url, &tmp_version);
 	} else {
-		error("FAILED TO VERIFY SIGNATURE OF %s. Operation proceeding due to \n"
-		      " --nosigcheck, but system security may be compromised\n",
-		      url);
-		journal_log_error("swupd security notice:  --nosigcheck used to bypass version signature");
+		warn("The --nosigcheck flag was used, THE SIGNATURE OF %s WILL NOT BE VERIFIED\n", url);
```
Comment (karthikprabhuvinod) [inline src/version.c]: Maybe we should stop using Capitalization

File: src/version.c [hunk 5]
```diff
@@ -176,10 +176,9 @@ static int get_version_from_url(char *url)
...
+		info("Be aware that this compromises the system security\n\n");
```
Comment (karthikprabhuvinod) [inline src/version.c]: I think we need this: info("Be aware that this compromises the system security\n"); You have two line feeds

File: src/signature.c (hunk 1 above)
Comment (castulo) [reply src/signature.c]: Well that typo in the sentence was not part of my commit, but sure, I can fix it.

File: src/signature.c (hunk 2 above)
Comment (castulo) [reply src/signature.c]: Same here

File: src/signature.c (hunk 3 above)
Comment (castulo) [reply src/signature.c]: I explained that in the commit message:

"When performing signature verification to the MoM or to the version
file, if there is any error during the process, a series of errors are
printed to stderr. Many of these error messages are duplicated and most
of them are too detailed.

This commit filters those detailed messages so they are only seen if the
user is using the --debug flag, showing only a few more general
error messages about the signature failure."

File: src/version.c (hunk 4 above)
Comment (castulo) [reply src/version.c]: yeah, I don't like the capitalization either, I didn't want to change it to lower case just so I was more conservative in my change, I can certainly change it to lowercase y @otaviobp is ok with that.

File: src/version.c (hunk 5 above)
Comment (castulo) [reply src/version.c]: those two lines "\n" are on purpose, I usually add two line breaks when I print important errors to highlight them. I think it looks nicer in the output when the error message is separated from the normal workflow.

**Comment by castulo** [comment]
> looks good

@otaviobp are you ok with converting those messages that right now are with UPPER CASE to be normal lower case? @karthikprabhuvinod and I agree it would look better, but wanted to double check with you

File: src/version.c (hunk 4 above)
Comment (castulo) [reply src/version.c]: I'll change that to normal lower case

**Comment by otaviobp** [comment]
> > looks good
> 
> @otaviobp are you ok with converting those messages that right now are with UPPER CASE to be normal lower case? @karthikprabhuvinod and I agree it would look better, but wanted to double check with you

Yes, I liked that. upper case looks like it's screaming.

File: src/version.c (hunk 4 above)
Comment (castulo) [reply src/version.c]: after some discussion, we decided to leave these messages as upper case

**Comment by castulo** [comment]
v2:
 - I just change "Failed open" with "Failed to open"

We decided to leave the certificate messages as uppercase.

**Pull request merged**